### ISSUE TITLE
[Launchpad] When getting a service fails, report more information

### ISSUE THREAD

**Issue by dleangen** [bndtools/bnd#3844]
It would be very useful if the method waitForService(), when failed, would report why the service was not returned.

In some cases at least, it is due to a missing @Reference. In this case, to quickly understand why the test failed, it would be very helpful to list the missing references.

**Comment by pkriens** [comment]
If the waitForService times out it tries to give a quite detailed report. I.e. if there is a service with the proper name but in another classloader this is reported in detail. 
Not sure what more to do.

**Comment by dleangen** [comment]
Are you referring to launchpad.report()? Or something else?

The launchpad.report() showed all bundles and components as being active, and shows a list of components.

From the perspective of somebody who was not party to designing and coding Launchpad, I looked everywhere I could think of, but found no such report, so I had to spend quite a bit of time combing manually through all my components and pick them apart. That manual work could have been avoided had I been aware of the report you are referring to.

Anyway, my tests are working now, so I will close the issue.
